### ISSUE TITLE
Ошибка при установке типа агента

### ISSUE THREAD

**Issue by Ilise-dot** [shtrih-m/fr_drv_ng#297]
Привет.
Возникла проблема при установке типа агента на C# под Linux.  ФФД 1.05. Для билда брал fr_drv_ng_linux_x86_64_1.6.0-21-ge5b15e7
Устанавливаю тип агента следующим образом:
```
// Тип агента
_drv.TagNumber = 1222;
_drv.TagType = 0;
_drv.TagValueInt = 64;
_drv.FNSendTagOperation();

// Данные агента
_drv.TagNumber = 1223;
_drv.TagType = 8;
_drv.FNBeginSTLVTag();
```
В этом случае, в методе FNSendTagOperation выдаст ошибку 115 (не поддерживается в данном режиме). Но на windows с этим проблем нет. 
Позиции чека печатаются несколько раз. И если будет >1 позиции, то в этом случае, в проходе по цикле после 1 уже ошибки в методе FNSendTagOperation не будет. И все позиции кроме одной будут с типом агента.
Если отправляем 1 позицию, то ошибка остается. Проверял в 2 утилитах, ошибка возникает на обеих...
Если перед установкой типа агента ставлю OpenCheck, то будет ошибка 94. 
_drv.CheckType = 1, пробовал с 0, та же самая ситуация.
Z отчет, X отчет печатает без проблем. Если печатать чек без типа агентов, то ошибок нет. 

Как быть? Или попробовать использовать не последнюю версию драйвера?

**Comment by olefard** [comment]
Нужен лог.

Похоже в вашем случае(работаете не с КЯ) `FNSendTagOperation` надо посылать после `FNOperation`. Возможно из-за этого и ошибка 115 и то, что все позиции кроме первой с агентом.

**Comment by Ilise-dot** [comment]
> Нужен лог.
> 
> Похоже в вашем случае(работаете не с КЯ) `FNSendTagOperation` надо посылать после `FNOperation`. Возможно из-за этого и ошибка 115 и то, что все позиции кроме первой с агентом.

Часть лога с одной из утилок. 
![image](https://github.com/user-attachments/assets/12782e1c-af1f-459d-86d8-ffeb03665f90)

Я написал сначала один комментарий (стер его часть), но после посмотрел свой код и понял, что если вызвать сначала FNOperation, а после задавать все теги, то ошибки и правда не будет. Т.к. такой вариант я как раз и пробовал. У меня раньше FNOperation стоял после FNSendTagOperation, а если поставить его в начале, то ошибки нет. 

UP
Но странно, что windows 10 аналогичный код (где FNSendTagOperation до FNOperation) и ошибки никакой нет.

**Comment by Ilise-dot** [comment]
> Нужен лог.
> 
> Похоже в вашем случае(работаете не с КЯ) `FNSendTagOperation` надо посылать после `FNOperation`. Возможно из-за этого и ошибка 115 и то, что все позиции кроме первой с агентом.

Да, вызов FNOperation до начала установки тегов в новой позиции, и уже последующий вызов FNSendTagOperation решает проблему. Спасибо за информацию!
Но почему на w10 порядок FNSendTagOperation ->FNOperation работает нормально, а на линуксе с данной либой нет - непонятно...

**Comment by olefard** [comment]
Если на виндовс и линукс работаете с одинаковой кассой, то поведение должно быть одинаковое.

Нужны логи с виндовс и линукс :) Тогда смогу сказать определенней, что происходит.

**Comment by Ilise-dot** [comment]
> Если на виндовс и линукс работаете с одинаковой кассой, то поведение должно быть одинаковое.
> 
> Нужны логи с виндовс и линукс :) Тогда смогу сказать определенней, что происходит.

ККМ одна. Под w10 и линукс разные сборки. Поэтому не критично)
